stockfish/models.py

```diff
@@ -552,10 +557,10 @@ def get_top_moves(self, num_top_moves: int = 5) -> List[dict]:
             elif (
                 ("multipv" in current_line)
                 and ("depth" in current_line)
-                and current_line[current_line.index("depth") + 1] == self.depth
```

Comment:
This is actually not needed, from what I understand. Since the line `bestmove` has appeared in the output, Stockfish is done searching (depending on multiple factors), and the top moves are listed at the bottom, indexed with `multipv`. 

Similiarly for `nodes`. The number of requested nodes may never be calulcated, in cases of mate in two, for example.